dvc/repo/experiments/__init__.py

```diff
@@ -362,6 +401,10 @@ def _update_params(self, params: dict):
             suffix = path.suffix.lower()
             modify_data = MODIFIERS[suffix]
             with modify_data(path, fs=self.repo.fs) as data:
+                new_params = self._get_new_params(data, params[params_fname])
+                if new_params:
+                    error_msg = self._format_new_params_msg(new_params, path)
+                    raise DvcException(error_msg)
```

Comment:
Rasing a new exception created for this case (`ParamDoesNotExistError`?) or an existing one (`MissingParamsError`? from `dvc.dependency.param`) would be probably better than raising the generic `DvcException`.